File: android/app/src/main/kotlin/com/kb/flutterkotlinmultiplatform/message/MessageConverter.kt
package com.kb.flutterkotlinmultiplatform.message

import com.kb.shared.presentation.message.MessageConverterInterface
import com.kb.shared.domain.model.MessageModel

class MessageConverter: MessageConverterInterface {

    override fun convertMessageToBuffer(messageModel: MessageModel): Any{
        return messageModel.toMessageModelProto().protoMarshal()
    }
}

File: ios/Runner/MessageConverter.swift
//
//  MessageConverter.swift
//  Runner
//
//  Created by Kweku on 12/01/2022.
//

import Foundation
import shared
import SwiftProtobuf

class MessageConverter: MessageConverterInterface {
    func convertMessageToBuffer(messageModel: MessageModel) -> Any {
        return MessageModelProto.with {
            $0.messageID = messageModel.messageId
            $0.title = messageModel.title
            $0.message = messageModel.message
        }.toFlutterStandardTypedData()
    }


}
